### PR TITLE
Environment_oM: Update/Upgrade

### LINKED ISSUE #417: Environment_oM: Material roughness
Materials should also specify their roughness in addition to a construction roughness. This is to be looked at in either March or April sprints pending the outcome of other material refactoring.

### LINKED ISSUE #457: Environment_oM: Map to new Physical/Analytical
With the addition of physical/analytical objects, and crucially, the discipline independent representations of a panel - revise the Environment oM and propagate through accordingly.

This issue will address:
 - Revising the oM for the discipline agnostic objects
 - Revising the oM structure and namespaces per discussions
 - Revising the oM data and ECS components per discussions

This will result in a fairly major change to Environment oM but this has been considered and accepted as required by the discipline at this time.

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #417 
Fixes #457 

Updates the Environment oM, restructures based on style changes discussed with @adecler and @al-fisher during workshop sessions.

Updates the Environment oM to utilise the new Panel and Opening definitions from Analytical and Physical oMs.

### Test files
N/A

#### Changed
 - Changed Environment oM to utilise new capability from Analytical oM
 - Changed Environment oM to utilise new capability from Physical oM
 - Changed Environment oM to new structure and style

### Additional comments
**Do not merge this PR until all dependent PRs are ready to merge - dependent PRs are now being worked on but this PR is being raised for visibility**

**Reviewers are invited to review for approving ready to merge when everything else is ready.**

**People who should be aware of this PR but are not being asked to review**
@tg359 
@ZiolkowskiJakub 
@mariatjader

### CODE REVIEW

File: Environment_oM/Elements/edge.cs [hunk 1]
```diff
@@ -20,23 +20,24 @@
...
 using System.Collections.Generic;
+using BH.oM.Geometry;
+using BH.oM.Base;
+using BH.oM.Analytical.Elements;
+using BH.oM.Environment.Properties;
 
 namespace BH.oM.Environment.Elements
 {
-    public class AirHandlingUnit : BHoMObject, IEquipment
+    public class Edge : BHoMObject, IEnvironmentObject, IEdge
```
[the remark this answers is not in the document]
Comment (FraserGreenroyd) [reply Environment_oM/Elements/edge.cs]: Done.

**Comment by al-fisher** [comment]
@FraserGreenroyd - referenced across your new good local prototyping of IBHoMFragment to parent https://github.com/BHoM/BHoM/issues/293 - for wider review and capture beyond close out of this PR

**Comment by FraserGreenroyd** [comment]
Hi @al-fisher - you are correct - the `IEnvironmentObject` is merely the prototyping element for `List<IBHoMFragment>`. If/when `IBHoMFragment` is moved up a level and is accessible to all BHoM objects (or all BHoM objects that are deemed to deserve them?), then the `IEnvironmentObject` will be removed, and instances of it in the engine will be replaced.

There is no desire at this time to continue using it beyond that purpose - hope that clarifies :smile:

**Comment by al-fisher** [comment]
> hope that clarifies

It certainly does @FraserGreenroyd ! We are on the same page! Thanks!

**Comment by al-fisher** [comment]
Thanks @FraserGreenroyd - just approved following your notes

**Pull request merged**